presto-spark-base/src/main/java/com/facebook/presto/spark/execution/PrestoSparkTaskExecutorFactory.java

```diff
@@ -493,7 +556,10 @@ private static OptionalLong computeAllSplitsSize(List<TaskSource> taskSources)
             BlockEncodingManager blockEncodingManager,
             OutputBufferMemoryManager memoryManager,
             DataSize targetAverageRowSize,
-            Optional<OutputPartitioning> preDeterminedPartition)
+            Optional<OutputPartitioning> preDeterminedPartition,
+            TempStorage tempStorage,
+            TempDataOperationContext tempDataOperationContext,
+            long maxBufferedBytes)
```

Comment:
`DataSize storageBasedBroadcastJoinWriteBufferSize`

Generally the rule of thumb that we have is we are trying to carry the `DataSize` / `Duration` objects all the way to the performance critical place, and only then we convert it to `int` or `long`